.github/workflows/tests.yml

```diff
@@ -0,0 +1,93 @@
+on:
+  - pull_request
+  - push
+
+name: Tests
+
+jobs:
+  unit:
+    name: Unit and Integration tests
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          # - windows-latest
+
+        php-version:
+          - "8.0"
+
+        dependencies:
+          - lowest
+          - highest
+
+    steps:
+      - name: Configure git to avoid issues with line endings
+        if: matrix.os == 'windows-latest'
+        run: git config --global core.autocrlf false
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          coverage: xdebug3
+          tools: none
+
+      - name: Install composer dependencies
+        uses: ramsey/composer-install@1.3.0
+        with:
+          dependency-versions: ${{ matrix.dependencies }}
+          composer-options: --no-ansi --no-interaction --no-progress
+
+      - name: Run tests with phpunit
+        env:
+          NOTION_TOKEN: ${{ secrets.NOTION_TOKEN }}
+        run: vendor/bin/phpunit --coverage-clover dist/phpunit/clover.xml
+
+      - name: Send code coverage report to Codecov.io
+        uses: codecov/codecov-action@v1
```

Comment:
One thing you may want to start doing instead of coverage, is mutation testing, with either of:

 * `infection/infection`
 * `roave/infection-static-analysis-plugin`

Both should give you a good idea of which areas in the code are not tested well enough, but a mutation test score of >80% is generally really good :) 

Code coverage itself is mostly a vanity metric these days: admirable, but not a clear indicator of tested/untested paths.

Reply:
Hi @mariosimao, I've got some learning days at work and I'm basically alternating reading non tech things and playing with mutation testing. If it's useful to you I can put together a pull request that adds mutation testing as suggested by @Ocramius. Looks like https://infection.github.io/ is a good way to go.

Reply:
Hi @WillGibson! It would be awesome to have mutation tests. Feel free to open a PR.

I haven't written any documentation about how to contribute yet. Basically, start from the `develop` branch and open the pull request with it as the target. I am trying to follow [GitFlow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).

Reply:
Cool. My next block of learning days is early December. I'll get onto it then if not before.

Reply:
https://github.com/mariosimao/notion-sdk-php/pull/52